x-pack/platform/plugins/shared/features/server/feature_registry.ts

```diff
@@ -269,7 +269,8 @@ export class FeatureRegistry {
               );
             }
 
-            if (referencedPrivilege.disabled) {
+            // Enabled privileges cannot be replaced with disabled ones.
+            if (referencedPrivilege.disabled && !privilege.disabled) {
```

Comment:
This makes sense but looks like the existing tests lack coverage for the following scenarios:
 - Disabled → Disabled privilege replacement
 - Disabled → Enabled privilege replacement

I don't think it's necessary to block this PR, so will create a follow-up issue